src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityAnalysisResultRepository.java

```diff
@@ -95,4 +134,257 @@ public String findStatus(UUID resultUuid) {
             return null;
         }
     }
+
+    interface SensitivityConsumer {
+        void consume(SensitivityEmbeddable sar, ContingencyEmbeddable c, SensitivityFactorP f);
+    }
+
+    private int apply(AnalysisResultEntity sas, Collection<String> funcIds, Collection<String> varIds, Collection<String> contingencyIds,
+        List<ContingencyEmbeddable> cs, List<SensitivityFactorP> fs, SensitivityFunctionType funcType, boolean beforeOverAfter,
+        Set<String> allFunctionIds, Set<String> allVariableIds, Set<String> allContingencyIds, SensitivityConsumer handle) {
+        int count = 0;
+        for (SensitivityEmbeddable sar : sas.getSensitivities()) {
+            int fi = sar.getFactorIndex();
+            SensitivityFactorP f = fs.get(fi);
+
+            int ci = sar.getContingencyIndex();
+            ContingencyEmbeddable c = ci < 0 ? null : cs.get(ci);
+            if ((ci < 0) == beforeOverAfter && f.getFunctionType() == funcType) {
+                count += 1;
+                if (allFunctionIds != null) {
```

Comment:
allFunctionIds and allVariableIds are never null ...

Reply:
I may help appease Sonar but I wonder why IntelliJ did tell me so.